library/waku_thread/inter_thread_communication/requests/peer_manager_request.nim

```diff
@@ -95,14 +97,29 @@ proc process*(
       return err($error)
     await waku.node.peerManager.disconnectNode(peerId)
     return ok("")
+  of DIAL_PEER:
+    let remotePeerInfo = parsePeerInfo($self[].peerMultiAddr).valueOr:
+      error "DIAL_PEER failed", error = $error
+      return err($error)
+    let conn = await waku.node.peerManager.dialPeer(remotePeerInfo, $self[].protocol)
+    if $self[].protocol != "" and conn.isNone():
+      let msg = "failed dialing peer"
+      error "DIAL_PEER failed", error = msg
+      return err(msg)
   of DIAL_PEER_BY_ID:
     let peerId = PeerId.init($self[].peerId).valueOr:
       error "DIAL_PEER_BY_ID failed", error = $error
       return err($error)
     let conn = await waku.node.peerManager.dialPeer(peerId, $self[].protocol)
-    if conn.isNone():
+    if $self[].protocol != "" and conn.isNone():
```

Comment:
How we should handle the condition of `$self[].protocol == "" and conn.isNone()`?
The same question applies to the `DIAL_PEER` operation:)